packages/world/src/EntityId.sol

```diff
@@ -3,21 +3,71 @@ pragma solidity >=0.8.24;
 
 import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
 import { Systems } from "@latticexyz/world/src/codegen/tables/Systems.sol";
+import { WorldContextConsumerLib } from "@latticexyz/world/src/WorldContext.sol";
 
+import { PlayerStatus } from "./codegen/tables/PlayerStatus.sol";
+import { ReversePlayer } from "./codegen/tables/ReversePlayer.sol";
 import { Program } from "./codegen/tables/Program.sol";
+import { ObjectType } from "./codegen/tables/ObjectType.sol";
 import { BaseEntity } from "./codegen/tables/BaseEntity.sol";
 import { Energy, EnergyData } from "./codegen/tables/Energy.sol";
 
+import { MovablePosition, Position } from "./utils/Vec3Storage.sol";
+import { updatePlayerEnergy, updateMachineEnergy } from "./utils/EnergyUtils.sol";
+import { getForceField } from "./utils/ForceFieldUtils.sol";
+
+import { checkWorldStatus } from "./Utils.sol";
+import { ObjectTypeId } from "./ObjectTypeId.sol";
+import { ObjectTypes } from "./ObjectTypes.sol";
+import { ObjectTypeLib } from "./ObjectTypeLib.sol";
+import { Vec3 } from "./Vec3.sol";
+import { MAX_ENTITY_INFLUENCE_HALF_WIDTH } from "./Constants.sol";
+
+using ObjectTypeLib for ObjectTypeId;
+
 type EntityId is bytes32;
 
 library EntityIdLib {
-  function isBaseEntity(EntityId self) internal view returns (bool) {
-    return EntityId.unwrap(BaseEntity._get(self)) == bytes32(0);
+  // We need to use this internal library function in order to obtain the msg.sig and msg.sender
```

Comment:
why do you need this if library functions are delegate called?

Reply:
because msg.data is not preserved in delegate calls, so you can't get the selector nor the msgSender due to how mud encodes the msg.sender in the calldata